moveit_core/kinematic_constraints/src/utils.cpp

```diff
@@ -273,3 +273,156 @@ moveit_msgs::Constraints kinematic_constraints::constructGoalConstraints(const s
 
   return goal;
 }
+
+bool kinematic_constraints::validatePositionConstraints(const robot_state::RobotState& state,
+                                                        moveit_msgs::Constraints& c)
+{
+  ROS_INFO("Validating position constraints.");
+
+  for (auto pos_con_it = c.position_constraints.begin(); pos_con_it != c.position_constraints.end(); ++pos_con_it)
+  {
+    ROS_INFO("Cycling through position constraint.");
+    ROS_INFO_STREAM("Current position constraint: link_name "
+                    << pos_con_it->link_name << ", offset " << pos_con_it->target_point_offset.x << " "
+                    << pos_con_it->target_point_offset.y << " " << pos_con_it->target_point_offset.z
+                    << ". header/frame_id: " << pos_con_it->header.frame_id);
+    // If the link is not found in the robot model
+    if (!state.hasLinkModel(pos_con_it->link_name))
+    {
+      // NOTE: This could use state.knowsFrameTransform() and state.getFrameTransform(), but those don't return the
+      // link_frame of the robot,
+      //       and I can't think of a good name for a function that returns bool frame_exists, Eigen::Affine3d transform
+      //       and std::string link_name.
+      std::vector<const robot_state::AttachedBody*> bodies;
+      state.getAttachedBodies(bodies);
+      Eigen::Affine3d t;
+      bool transform_found = false;
+      std::string robot_link_name;
+      // Try finding it in attached bodies
+      if (state.hasAttachedBody(pos_con_it->link_name))
+      {
+        ROS_INFO_STREAM("State has attached body with name " << pos_con_it->link_name);
+        auto ab = state.getAttachedBody(pos_con_it->link_name);
+        t = ab->getFixedTransforms()[0];
+        robot_link_name = ab->getAttachedLinkName();
+        transform_found = true;
+      }
+      else  // Try finding it in attached bodies' named frames
+      {
+        ROS_INFO_STREAM("State has no attached body with name " << pos_con_it->link_name << ". Checking named frames.");
+        for (auto ab : bodies)
+        {
+          ROS_INFO_STREAM("Checking attached body " << ab->getName());
+          ROS_INFO_STREAM(ab->getName() << " has " << ab->getNamedTransforms().size() << " named frames.");
+          if (ab->hasNamedTransform(pos_con_it->link_name))
+          {
+            t = ab->getNamedTransform(pos_con_it->link_name);
+            robot_link_name = ab->getAttachedLinkName();
+            ROS_INFO_STREAM("State has "
+                            << " named frame called " << pos_con_it->link_name << " on attached body "
+                            << ab->getName());
+            transform_found = true;
+          }
+        }
+      }
+
+      if (transform_found)
+      {
```

Comment:
Remove the else and just use:

      if (!transform_found)
           return false